actors/evm/src/interpreter/execution.rs

```diff
@@ -181,222 +159,236 @@ impl<'r, 'a, RT: Runtime + 'r> Machine<'r, 'a, RT> {
                 break;
             }
 
-            match self.step()? {
-                ControlFlow::Continue => {
-                    self.pc += 1;
-                }
-                ControlFlow::Jump => {}
-                ControlFlow::Exit => {
-                    break;
-                }
-            };
+            self.step();
+
+            if self.exit.is_some() {
+                break;
+            }
```

Comment:
Hm. Can we move this condition to the top to reduce the number of breaks?

I.e.:

```rust
while self.exit.is_none() && self.pc < self.bytecode.len() {
    self.step();
}
```

Reply:
ok, somewhat prettier; my way is more explicit however

Reply:
Some more thoughts here: the current code is optimal in terms of branches.

Initially, exit is known to be None (initial state invariant), but the code may be empty so we need to check pc against the bound.
Then we execute the instruction, which may change exit, at which point we check it and loop.

No matter how you look at it, the prettier variant will execute one more check/possible branch than the current code. 

So i suggest we keep it as is.